### PR TITLE
add chat support

### PR DESCRIPTION
**Pull request by bugsz**
Added a minimal runnable demo for live chat with LLM model

## Implemented
+ Select scenarios and agents within predefined sets in the database
+ Run human-model interaction
+ Stop the interaction and evaluate automatically

## Issues
+ Hard to do type annotations for the `st.session_state`.

### CODE REVIEW

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]
Meta comments:

* update the readme (one should be able to reproduce your features through the readme)
* get rid of all the things related to haicosystem, including names, pics and stuff. Change them to another name. I am honestly thinking about making this a package, so maybe let's call it `socialstream` tentatively?
* move all the package related stuff to the folder `socialstream`

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]
Run 
`streamlit run app.py`

I got

```
2024-08-05 15:53:20.069 Uncaught app exception
Traceback (most recent call last):
  File "/Users/xuhuizhou/miniconda3/envs/socialstream/lib/python3.11/site-packages/streamlit/runtime/scriptrunner/exec_code.py", line 85, in exec_func_with_error_handling
    result = func()
             ^^^^^^
  File "/Users/xuhuizhou/miniconda3/envs/socialstream/lib/python3.11/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 576, in code_to_exec
    exec(code, module.__dict__)
  File "/Users/xuhuizhou/Web_dev/project_webs/sotopia-demo/app.py", line 6, in <module>
    from socialstream.chat import chat_demo, initialize_session_state
  File "/Users/xuhuizhou/Web_dev/project_webs/sotopia-demo/socialstream/chat/__init__.py", line 1, in <module>
    from .chat import chat_demo, initialize_session_state
  File "/Users/xuhuizhou/Web_dev/project_webs/sotopia-demo/socialstream/chat/chat.py", line 9, in <module>
    from sotopia.envs.evaluators import (
ImportError: cannot import name 'EvaluationForTwoAgents' from 'sotopia.envs.evaluators' (/Users/xuhuizhou/miniconda3/envs/socialstream/lib/python3.11/site-packages/sotopia/envs/evaluators.py)
```

**Comment by XuhuiZhou** [comment]
> Run `streamlit run app.py`
> 
> I got
> 
> ```
> 2024-08-05 15:53:20.069 Uncaught app exception
> Traceback (most recent call last):
>   File "/Users/xuhuizhou/miniconda3/envs/socialstream/lib/python3.11/site-packages/streamlit/runtime/scriptrunner/exec_code.py", line 85, in exec_func_with_error_handling
>     result = func()
>              ^^^^^^
>   File "/Users/xuhuizhou/miniconda3/envs/socialstream/lib/python3.11/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 576, in code_to_exec
>     exec(code, module.__dict__)
>   File "/Users/xuhuizhou/Web_dev/project_webs/sotopia-demo/app.py", line 6, in <module>
>     from socialstream.chat import chat_demo, initialize_session_state
>   File "/Users/xuhuizhou/Web_dev/project_webs/sotopia-demo/socialstream/chat/__init__.py", line 1, in <module>
>     from .chat import chat_demo, initialize_session_state
>   File "/Users/xuhuizhou/Web_dev/project_webs/sotopia-demo/socialstream/chat/chat.py", line 9, in <module>
>     from sotopia.envs.evaluators import (
> ImportError: cannot import name 'EvaluationForTwoAgents' from 'sotopia.envs.evaluators' (/Users/xuhuizhou/miniconda3/envs/socialstream/lib/python3.11/site-packages/sotopia/envs/evaluators.py)
> ```

Okay I fixed this

**Comment by XuhuiZhou** [comment]
The chat mode is not working for me. The observation is loading forever but no error msg from the console.

File: .streamlit/config.toml
Comment (XuhuiZhou) [inline .streamlit/config.toml:1]: Why remove this? This is the place to variables to control streamlit behavior. Please also illustrate this in the documentation as well. I.e., how people should set various env variables.

File: app.py [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
-episode_number = st.text_input("Enter episode number:", value="2")
-episode = EpisodeLog.find(EpisodeLog.tag == "haicosystem_debug")[int(episode_number)]  # type: ignore
-assert isinstance(episode, EpisodeLog)
-messages = render_for_humans(episode)
-streamlit_rendering(messages)
\ No newline at end of file
+from socialstream.chat import chat_demo, initialize_session_state
+from socialstream.rendering import rendering_demo
+
+# os.environ["REDIS_OM_URL"] = st.secrets["REDIS_OM_URL"]
```
Comment (XuhuiZhou) [inline app.py]: We should keep this as it would be necessary for deployment

File: pyproject.toml [hunk 2]
```diff
@@ -1,23 +1,21 @@
 [tool.poetry]
-name = "haicosystemDemo"
+name = "socialstream"
 version = "0.0.1"
 description = "An demo"
-authors = ["Xuhui Zhou <zhouxuhui2018@gmail.com>"]
+authors = ["Zhe Su <suzhesz00@gmail.com>"]
```
Comment (XuhuiZhou) [inline pyproject.toml]: also include my name :)

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]

File: .streamlit/config.toml
Comment (bugsz) [reply .streamlit/config.toml:1]: Not sure what happened there. If I specified the template there is an error `OSError: [Errno 28] inotify watch limit reached`.

File: pyproject.toml (hunk 2 above)
Comment (bugsz) [reply pyproject.toml]: Oh I see. I was assuming there could be only one author!

File: .streamlit/config.toml
Comment (XuhuiZhou) [reply .streamlit/config.toml:1]: Try see if it's other codes in the repo conflicting this. But yeah, we don't want we can not control the config file

File: app.py (hunk 1 above)
Comment (XuhuiZhou) [reply app.py]: Okay, actually remove this line

File: socialstream/utils.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+class messageForRendering(TypedDict):
+    role: str
+    type: str
+    content: str
+
+
+def parse_reasoning(reasoning: str, num_agents: int) -> tuple[list[str], str]:
+    """Parse the reasoning string into a dictionary."""
+    sep_token = "SEPSEP"
```
Comment (XuhuiZhou) [inline socialstream/utils.py:14]: what's this?

File: socialstream/chat/chat.py
Comment (XuhuiZhou) [inline socialstream/chat/chat.py:1]: remove this file?

File: socialstream/utils.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            .strip("\n")
+        )
+    all_chunks = reasoning.split(sep_token)
+    general_comment = all_chunks[0].strip(" ").strip("\n")
+    comment_chunks = all_chunks[-num_agents:]
+
+    return comment_chunks, general_comment
+
+
+def render_for_humans(episode: EpisodeLog) -> list[messageForRendering]:
```
Comment (XuhuiZhou) [inline socialstream/utils.py:28]: It's best if we leave this function to sotopia instead of here.

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]
Delete haicosystemDemo? 
Also change the chat component display order, use the old sotopia-space as the reference?

File: socialstream/utils.py (hunk 3 above)
Comment (bugsz) [reply socialstream/utils.py:14]: I just copy and paste from the haicosystem database (?

**Comment by bugsz** [comment]
Seems it is not in the database right now

File: socialstream/utils.py (hunk 3 above)
Comment (XuhuiZhou) [reply socialstream/utils.py:14]: ah okay, let's leave it there for now.

File: socialstream/utils.py (hunk 4 above)
Comment (XuhuiZhou) [reply socialstream/utils.py:28]: let's leave it there for now

File: app.py (hunk 1 above)
Comment (XuhuiZhou) [reply app.py]: In the readme, tell people they need to add secrets if they want to run things locally, for both redis_om and open_ai key. Streamlit automatically adds secrets to the env variable

**Comment by XuhuiZhou** [comment]
> Seems it is not in the database right now

?

**Comment by bugsz** [comment]
My bad. I mean codebase

File: socialstream/utils.py (hunk 4 above)
Comment (bugsz) [reply socialstream/utils.py:28]: but we do have a separate `render_for_humans` in sotopia?

**Comment by XuhuiZhou** [comment]
> My bad. I mean codebase

okay, that's good

File: socialstream/utils.py (hunk 4 above)
Comment (XuhuiZhou) [reply socialstream/utils.py:28]: hmmm, let's leave it there for now.

File: socialstream/chat/chat.py [hunk 5]
```diff
@@ -159,242 +225,334 @@ def get_env_agents(env_agent_combo) -> None:
...
+            )
+        with agent_col2:
+            agent_choice_2 = st.selectbox(
+                "Choose the second agent:",
+                agent_list_2.keys(),
+                disabled=st.session_state.active,
+                index=1,
+            )
+        print(agent_choice_1, agent_choice_2)
+        user_position = st.selectbox(
```
Comment (bugsz) [inline socialstream/chat/chat.py:263]: Use Agent 1/2 or Agent 1/2's name as content?

File: socialstream/chat/chat.py [hunk 6]
```diff
@@ -159,242 +225,334 @@ def get_env_agents(env_agent_combo) -> None:
...
-            "Choose the second agent:",
-            agent_list_2.keys(),
-            disabled=st.session_state.active,
+
+        agent1_col, agent2_col = st.columns(2)
+        agent_cols = [agent1_col, agent2_col]
+        for agent_idx, agent_info in enumerate(agent_infos):
+            agent_col = agent_cols[agent_idx]
+            with agent_col:
+                st.markdown(
```
Comment (bugsz) [inline socialstream/chat/chat.py:311]: st.markdown (pure text but with format) or st.text_area (with border, but no md format)?

File: socialstream/chat/chat.py (hunk 5 above)
Comment (XuhuiZhou) [reply socialstream/chat/chat.py:263]: We are not using this script anymore right?

File: socialstream/chat/chat.py (hunk 6 above)
Comment (XuhuiZhou) [reply socialstream/chat/chat.py:311]: ？are we using chat_new or chat?

Comment (bugsz) [reply socialstream/chat/chat.py:311]: I remove chat.py and rename chat_new.py

File: socialstream/chat/chat.py (hunk 5 above)
Comment (bugsz) [reply socialstream/chat/chat.py:263]: That is the previous chat_new.py as I rename it

**Comment by XuhuiZhou** [comment]
I think we could merge this now.

**Review by XuhuiZhou — APPROVED** [review]

**Pull request merged**